scripts/check_names.py

```diff
@@ -1016,8 +1164,21 @@ def main():
     log.addHandler(logging.StreamHandler())
 
     try:
-        code_parser = CodeParser(log)
-        parse_result = code_parser.comprehensive_parse()
+        if build_tree.looks_like_tf_psa_crypto_root(os.getcwd()):
+            tf_psa_crypto_code_parser = TFPSACryptoCodeParser(log)
+            parse_result = tf_psa_crypto_code_parser.comprehensive_parse()
+        elif build_tree.looks_like_mbedtls_root(os.getcwd()):
+            # Mbed TLS uses TF-PSA-Crypto, so we need to parse TF-PSA-Crypto too
+            tf_psa_crypto_code_parser = TFPSACryptoCodeParser(log)
+            tf_psa_crypto_parse_result = tf_psa_crypto_code_parser.comprehensive_parse()
+            mbedtls_code_parser = MBEDTLSCodeParser(log)
+            mbedtls_parse_result = mbedtls_code_parser.comprehensive_parse()
+            # Combine parse results together for NameChecker
+            parse_result = {}
+            for key in tf_psa_crypto_parse_result:
+                parse_result[key] = tf_psa_crypto_parse_result[key] + mbedtls_parse_result[key]
+        else:
+            raise Exception("This script must be run from Mbed TLS or TF-PSA-Crypto root")
```

Comment:
Since CodeParser class already checks if the script is being run from the proper folder in its `_init_()` function, what about removing the `else` branch and simplifying the rest as follows:
```suggestion
        # Mbed TLS uses TF-PSA-Crypto, so we parse TF-PSA-Crypto in both cases
        tf_psa_crypto_code_parser = TFPSACryptoCodeParser(log)
        parse_result = tf_psa_crypto_code_parser.comprehensive_parse()

        if build_tree.looks_like_mbedtls_root(os.getcwd()):
            mbedtls_code_parser = MBEDTLSCodeParser(log)
            mbedtls_parse_result = mbedtls_code_parser.comprehensive_parse()
            # Combine parse results together for NameChecker
            parse_result = {}
            for key in tf_psa_crypto_parse_result:
                parse_result[key] = tf_psa_crypto_parse_result[key] + mbedtls_parse_result[key]
```
?

Reply:
Thanks for the suggestion. I have refactored the code :)